### PR TITLE
Change: [Actions] Use -fuse-ld=lld for MinGW

### PR DESCRIPTION
**Pull request by glx22**
## Motivation / Problem
Linking is super slow for MinGW.

## Description
Try to speedup linking for MinGW using another linker.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -400,12 +402,12 @@ jobs:
...
 
         echo "::group::CMake"
-        cmake .. -G"MSYS Makefiles"
+        cmake .. -G"Ninja" -DCMAKE_CXX_FLAGS="-fuse-ld=lld"
         echo "::endgroup::"
 
         echo "::group::Build"
         echo "Running on $(nproc) cores"
-        cmake --build . -j $(nproc)
+        NINJA_STATUS="[%f/%t -- %e ]" cmake --build . -j $(nproc)
```
Comment (LordAro) [inline .github/workflows/ci-build.yml]:
```suggestion
        NINJA_STATUS="[%f/%t -- %e] " cmake --build . -j $(nproc)
```

**Comment by glx22** [comment]
[CI run times](https://github.com/OpenTTD/OpenTTD/actions/workflows/ci-build.yml) are so inconsistent, it's hard to see if this PR really improves things.

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -320,6 +320,8 @@ jobs:
 
     - name: Build
       shell: bash
+      env:
+        NINJA_STATUS: "[%f/%t -- %e] "
```
Comment (TrueBrain) [inline .github/workflows/ci-build.yml]: Maybe a comment what this does exactly?

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -395,17 +399,22 @@ jobs:
...
+        cmake .. \
+          -GNinja \
+          -DCMAKE_CXX_FLAGS="-fuse-ld=lld"
+          # EOF
         echo "::endgroup::"
 
         echo "::group::Build"
         echo "Running on $(nproc) cores"
-        cmake --build . -j $(nproc)
+        cmake --build .
```
Comment (TrueBrain) [inline .github/workflows/ci-build.yml:416]: no more -j? And if that was intentional, please update the `echo` above it too :)

Comment (glx22) [reply .github/workflows/ci-build.yml:416]: From https://ninja-build.org/manual.html#_running_ninja: `Ninja defaults to running commands in parallel anyway, so typically you don’t need to pass -j`

File: .github/workflows/ci-build.yml [hunk 4]
```diff
@@ -395,24 +399,28 @@ jobs:
...
+        NINJA_STATUS: "[%f/%t -- %e] " # [finished_edges/total_edges -- elapsed_time], default value is "[%f/%t] "
       run: |
         mkdir build
         cd build
 
         echo "::group::CMake"
-        cmake .. -G"MSYS Makefiles"
+        cmake .. \
+          -GNinja \
+          -DCMAKE_CXX_FLAGS="-fuse-ld=lld"
```
Comment (TrueBrain) [inline .github/workflows/ci-build.yml]: Missing \ at the end; otherwise the EOF marker isn't useful  :)

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**